lib/wasi.js

```diff
@@ -93,12 +104,30 @@ class WASI {
     this[kStarted] = true;
     this[kSetMemory](memory);
 
-    if (exports._start)
-      exports._start();
-    else if (exports.__wasi_unstable_reactor_start)
-      exports.__wasi_unstable_reactor_start();
+    try {
+      if (exports._start)
```

Comment:
since we can't easily validate the types of these functions (i guess we could link them to a stub module but that seems like a lot of effort), instead of returning from them i'd add a `return 0` below.